src/main/scala/is/hail/keytable/KeyTable.scala

```diff
@@ -127,10 +127,35 @@ object KeyTable {
 
     KeyTable(hc, rdd, struct, Array("ID"))
   }
+
+  def apply(hc: HailContext, rdd: RDD[Row], signature: TStruct, key: Array[String] = Array.empty): KeyTable = {
+    val rdd2 = rdd.mapPartitions { it =>
+      val region = MemoryBuffer()
+      val rvb = new RegionValueBuilder(region)
+      val rv = RegionValue(region)
+      it.map { row =>
+        region.clear()
+        rvb.start(signature)
+        rvb.addAnnotation(signature, row)
+        rv.setOffset(rvb.end())
+        rv
+      }
+    }
+    new KeyTable(hc, rdd2, signature, key)
+  }
 }
 
-case class KeyTable(hc: HailContext, rdd: RDD[Row],
-  signature: TStruct, key: Array[String] = Array.empty) {
+class KeyTable(val hc: HailContext,
+  val rdd2: RDD[RegionValue],
```

Comment:
Main purpose of change: KeyTable should contain RDD[RegionValue] instead of RDD[Row].   Follow pattern in VariantSampleMatrix: new constructor that converts from RDD[Row], lazy val rdd converts to Row, rdd now deprecated.